docs/specs.md

```diff
@@ -0,0 +1,316 @@
+Entity is described by entity spec. Following is an example of an entity spec
+
+```
+name: driver
+description: GO-JEK’s 2 wheeler driver
+tags:
+  - two wheeler
+  - gojek
+```
+
+### Attributes Supported In Entity Spec
+
+| Name | Type | Convention | Description |
+| ----- |----------| -----------| ------------|
+| `name`  | string   | Lower snake case (e.g driver or driver_area) | Entity name MUST be unique |
+| `description` | string | N.A. | Description of the entity |
+| `tags` | List of string | N.A. | Free form grouping |
+
+
+
+## Feature Spec
+A Feature is an individual measurable property or characteristic of an Entity. A feature has many to one relationship with an entity by referencing the entity's name from its feature spec. In the context of Feast, a Feature has the following attributes: 
+
+1. Entity - it must be associated with a known Entity to Feast (see Entity Spec)
+2. ValueType - the feature type must be defined, e.g. String, Bytes, Int64, Int32, Float etc.
+3. Data stores - for both serving and training must be defined
+
+Following is en example of feature spec
+
+```
+id: titanic_passenger.alone
+name: alone
+entity: titanic_passenger
+owner: zhiling.c@go-jek.com
+description: binary variable denoting whether the passenger was alone on the titanic.
+valueType:  INT64
+uri: http://jupyter.s.ds.golabs.io/user/your_user/lab/tree/shared/zhiling.c/feast_titanic/titanic.ipynb
+dataStores:
+  serving:
+    id: REDIS1
+  warehouse:
+    id: BIGQUERY1
+```
+
+### Attributes Supported In Feature Spec
+
+|Name|Type|Convention|Description|
+|:----:|----|----------|-----------|
+|`entity`|string|lower snake case (e.g. `driver`, `driver_area`)| Entity related to this feature|
+|`id`|string|lower snake case with format `[entity].[featureName]` (e.g.: `customer.age_prediction`) | feature id is a unique identifier of a feature, it is used both for feature retrieval from serving / warehouse|
+|`name`|string|lower snake case, e.g: `age_prediction`| short name of a feature |
+|`owner`|string|use email or any unique identifier| `owner` is mainly be used to inform user who is responsible of maintaining the feature |
+|`description`|string| Keep the description concise, more information about the feature can be provided as documentation linked by `uri` | human readable description of a feature |
+|`valueType`| Enum(one of: `BYTES`,`STRING`,`INT32`,`INT64`,`DOUBLE`,`FLOAT`, `BOOL`, `TIMESTAMP`)| N.A.| Value type of the feature |
+|`group`| string | lower snake case | feature group inherited by this feature.|
+| `tags` | List of string | N.A. | Free form grouping |
+| `options` | Key value string | N.A. | Option is used for extendability of feature spec|
+| `dataStores`| See below table | See below table | Storage used to store the feature|
+
+#### DataStores 
```

Comment:
declaration of datastores has been deprecated as of pr #159

Reply:
Oh great, I didn't know that. Does it make sense to remove this from python SDK's `Importer` API as well?

Reply:
Yes I think we should